framework/helpers/BaseJson.php

```diff
@@ -147,17 +148,14 @@ protected static function handleJsonError($lastError)
             return;
         }
 
-        $availableErrors = [];
```

Comment:
That won't works at all. Correct code is the following:

```php
     protected static function handleJsonError($lastError)
    {
        if ($lastError === JSON_ERROR_NONE) {
            return;
        }

        if (function_exists('json_last_error_msg')) {
            throw new InvalidArgumentException(json_last_error_msg(), $lastError);
        }

        if (array_key_exists($lastError, static::$jsonErrorMessages)) {
            throw new InvalidArgumentException(static::$jsonErrorMessages[$lastError], $lastError);
        }

        throw new InvalidArgumentException('Unknown JSON encoding/decoding error.');
    }

```

Reply:
1. `json_last_error_msg` is available since PHP 5.5 so for this case we don't need fallback at all.
2. We need fallback for PHP < 5.5 so we can delete all the fallback values that are 5.5+.
3. We can't use new constants since these are not defined by PHP.

Reply:
@samdark `static::$jsonErrorMessages` use constant names as keys, `$lastError` is numeric code of error